tests/unit/lib/SGMatrix_unittest.cc

```diff
@@ -761,3 +761,38 @@ TEST(SGMatrixTest,iterator)
 	for (auto v: mat)
 		EXPECT_EQ(mat[index++], v);
 }
+
+TEST(SGMatrixTest,column_iterator)
+{
+	constexpr index_t size=5;
+	SGMatrix<float64_t> mat(size, size);
+	linalg::range_fill(mat, 1.0);
+
+	auto begin_col = mat.begin_column();
+	auto end_col = mat.end_column();
+
+	EXPECT_EQ(mat.num_cols, std::distance(begin_col, end_col));
+	EXPECT_EQ(mat.get_column(0),*begin_col);
+	++begin_col;
+	EXPECT_EQ(mat.get_column(1),*begin_col);
+	--begin_col;
+	EXPECT_EQ(mat.get_column(0),*begin_col++);
+	EXPECT_TRUE(begin_col != end_col);
+	++begin_col;
+	EXPECT_EQ(mat.get_column(2),*begin_col--);
+	--begin_col;
+	EXPECT_EQ(mat.get_column(0),*begin_col);
+	begin_col += 2;
+	EXPECT_EQ(mat.get_column(2),*begin_col);
+	begin_col -= 1;
+	EXPECT_EQ(mat.get_column(1),*begin_col);
+	EXPECT_EQ(mat.get_column(2),begin_col[1]);
+	
+	auto new_itr = begin_col + 2;
+	EXPECT_EQ(mat.get_column(3),*new_itr);
+
+	//range-based loop should work as well
+	int idx=0;
+	std::for_each(mat.begin_column(),mat.end_column(),[&idx,mat](const auto& vec){ EXPECT_EQ(vec,mat.get_column(idx++));}); 
```

Comment:
doesn't this cover all the test above?

Reply:
Hm, we do want to check if += , -= are working, right?

Reply:
+= would be handled by for_each no? Anyway, I guess more tests can’t hurt!

Reply:
Haha yea, that's true. I meant += d btw. for_each increases by one only.And then there's also the deferencing one ```[]```